integration-tests/src/test/java/oracle/kubernetes/operator/ShutdownOptionsBase.java

```diff
@@ -0,0 +1,165 @@
+  
+
+  protected Domain createDomain(String domainNS, Map<String, Object> shutdownProps) throws Exception {
+
+    Map<String, Object> domainMap = createDomainMap(getNewSuffixCount(), "itpodsshutdown");
+    domainMap.put("namespace", domainNS);
+    domainMap.put("domainUID", domainNS + getNewSuffixCount());
+    domainMap.put("initialManagedServerReplicas", new Integer("1"));
+    domainMap.put("shutdownOptionsOverrides",shutdownProps);
+    LoggerHelper.getLocal().log(Level.INFO, "Creating and verifying the domain creation with domainUid: " + domainNS);
+    Domain domain = TestUtils.createDomain(domainMap);
+    domain.verifyDomainCreated();
+    return domain;
+  }
+
+  /**
+   * Send request to web app deployed on wls.
+   *
+   * @param testAppPath - URL path for webapp
+   * @param domain      - Domain where webapp is deployed
+   * @throws Exception  - Fails if can't find expected output from app.
+   */
+  protected static void callWebApp(String testAppPath, Domain domain)
+      throws Exception {
+
+    if (!domain.getLoadBalancerName().equalsIgnoreCase("NONE")) {
+      // url
+      StringBuffer testAppUrl = new StringBuffer("http://");
+      testAppUrl.append(domain.getHostNameForCurl());
+      testAppUrl.append("/");
+      if (domain.getLoadBalancerName().equals("APACHE")) {
+        testAppUrl.append("weblogic/");
+      }
+      testAppUrl.append(testAppPath);
+      // curl cmd to call webapp
+
+      StringBuffer webServiceUrl = new StringBuffer("curl --silent --noproxy '*' ");
+      webServiceUrl
+          .append(" -H 'host: ")
+          .append(domain.getDomainNs())
+          .append(".org' ")
+          .append(testAppUrl.toString());
+
+      // Send a HTTP request to keep open session
+      String curlCmd = webServiceUrl.toString();
+      ExecCommand.exec(curlCmd);
+    }
+  }
+
+  /**
+   * Shutdown managed server and returns spent shutdown time.
+   *
+   * @throws Exception If failed to shutdown the server.
+   */
+  protected static long shutdownServer(String serverName, String domainNS, String domainUid) throws Exception {
+    long startTime;
+    startTime = System.currentTimeMillis();
+    String cmd = "kubectl delete pod " + domainUid + "-" + serverName + " -n " + domainNS;
+    LoggerHelper.getLocal().log(Level.INFO, "command to shutdown server <" + serverName + "> is: " + cmd);
+    ExecResult result = ExecCommand.exec(cmd);
+    long terminationTime = 0;
+    if (result.exitValue() != 0) {
+      throw new Exception("FAILURE: command " + cmd + " failed, returned " + result.stderr());
+    }
+    TestUtils.checkPodCreated(domainUid + "-" + serverName, domainNS);
+    long endTime = System.currentTimeMillis();
+    terminationTime = endTime - startTime;
+    return terminationTime;
+  }
+
+  /**
+   * Check the applied properties are set in the pod.
+   *
+   * @param podName - name of the pod
+   * @param domainNS - domain namespace
```

Comment:
probably should call this podNS, or just namespace, since you are talking about a pod here, not a domain - confusing